### PR TITLE
Sidebar improvements and fixes. Modal improvements.

### PR DESCRIPTION
**Pull request by ISp1rit**

### CODE REVIEW

File: packages/sidebar/src/Sidebar.tsx [hunk 1]
```diff
@@ -168,51 +157,46 @@ export const PureSidebar = ({
...
-              >
-                <Scrollbar
-                  animate={ transformParams.animate }
-                  width={ rightWidth }
-                  marginLeft="-10px"
+              <ChildWrapper animate={ animate }>
+                <Scrollbars
+                  style={ {
+                    width: isResponsive ? width : maxWidth,
+                    transition: isResponsive && animate ? '0.5s' : 'none',
```
Comment (tundraray) [inline packages/sidebar/src/Sidebar.tsx]: Why isResponsive disable animation?

File: packages/sidebar/src/Sidebar.tsx [hunk 2]
```diff
@@ -168,51 +157,46 @@ export const PureSidebar = ({
...
-                style={ { width: transformParams.width } }
-                animate={ transformParams.animate }
-              >
-                <Scrollbar
-                  animate={ transformParams.animate }
-                  width={ rightWidth }
-                  marginLeft="-10px"
+              <ChildWrapper animate={ animate }>
+                <Scrollbars
+                  style={ {
```
Comment (tundraray) [inline packages/sidebar/src/Sidebar.tsx]: There are some problems: 
1) Here Scrollbars will be taking a new object of style in every render cycle
2) The style attribute is not god when styled-components is used.  https://styled-components.com/docs/api#attrs

File: packages/sidebar/src/Sidebar.tsx [hunk 3]
```diff
@@ -21,116 +23,105 @@ import {
...
 
 
-export const PureSidebar = ({
+export const PureSidebar: React.FC<ISidebarProps> = ({
   theme,
   children,
   navComponent,
   showScrollbar,
   arrowComponent = <Arrow />,
+  isResponsive = false,
```
Comment (tundraray) [inline packages/sidebar/src/Sidebar.tsx]: I'm not sure, but I think it's possible without this parameter. If children element have fixed width then will be work similar

File: packages/sidebar/src/Sidebar.tsx [hunk 4]
```diff
@@ -21,116 +23,105 @@ import {
...
-    width: maxWidth,
-    animate: false,
-    arrowToRight: false,
-  });
-
-  const [antiSelectLayer, changeAntiSelectLayer] = useState(false);
-  const sidebarRef = useRef<HTMLDivElement>(null);
-
-  const clickX = useRef(0);
+  const [width, setWidth] = useState(maxWidth);
```
Comment (tundraray) [inline packages/sidebar/src/Sidebar.tsx]: You made some state and you use them together. I think pointless refactoring :) What you think about it?

Comment (ISp1rit) [reply packages/sidebar/src/Sidebar.tsx]: 1) It is not obvious what is it transformParams, and it is hard and wrong to combine incombineble.
2) With current changes it is pretty clear what must be changed in concrete case.

File: packages/sidebar/src/Sidebar.tsx (hunk 2 above)
Comment (ISp1rit) [reply packages/sidebar/src/Sidebar.tsx]: If wrap Scrollbars inside styled we'll get console error about passing unrecognized props inside DOM element.
In current situation width always will change so there are no reason wrap it in useMemo.
And that is why current solution is OK.

File: packages/sidebar/src/Sidebar.tsx (hunk 3 above)
Comment (ISp1rit) [reply packages/sidebar/src/Sidebar.tsx]: Discussed that isResponsive is unnecessary in our projects so i removed it.

**Pull request merged**